dtd-catalog-exporter/src/services/mongodb-e-service.service.ts

```diff
@@ -0,0 +1,131 @@
+import { MongoClient } from "mongodb";
+import { env } from "../configs/env.js";
+import { EService, eserviceSchema, Attribute, attributeSchema, Tenant, tenantSchema } from "../models/index.js";
+
+export class MongoDBEServiceClient {
+  private client: MongoClient;
+
+  private constructor(client: MongoClient) {
+    this.client = client;
+  }
+
+  /**
+   * Connects to the mongodb database
+   */
+  public static async connect() {
+    const connectionString = `mongodb://${env.READ_MODEL_DB_USER}:${env.READ_MODEL_DB_PASSWORD}@${env.READ_MODEL_DB_HOST}:${env.READ_MODEL_DB_PORT}?directConnection=true&readPreference=secondaryPreferred&retryWrites=false`;
+    const client = await new MongoClient(connectionString).connect();
+    return new MongoDBEServiceClient(client);
+  }
+
+  /**
+   * Fetches all active e-services from the database, validates them and returns them.
+   * The e-services is considered active if it has at least one descriptor with state "Published" or "Suspended".
+   *
+   * @returns The array of e-services
+   */
+  async getEServices() {
+    return await this.client
+      .db(env.READ_MODEL_DB_NAME)
+      .collection<EService>(env.ESERVICES_COLLECTION_NAME)
+      .aggregate([
```

Comment:
note that `find` is lighter and has better performance, it is preferable when possible.
In this case I guess this could be refactored as
```
.find({"data.descriptors.state": {$in: ['Published', 'Suspended']}})
```
(not sure about the `replaceRoot` though)